### DISCUSSION TITLE
Excluding comprehension values

### DISCUSSION THREAD

**Discussion by dylankb** [cue-lang/cue#2021] [Q&A] [answered]
I would like to create a list of values from another set of data (e.g. another list) but exclude one value.  E.g. `cars` includes "tacoma" but `cars2` should exclude it.

```
cars: ["tacoma", "escort"]

cars2: [ for c in cars {
		if c != "tacoma" {c}
	}
]

// result
cars: ["tacoma", "escort"]
cars2: [{}, "escort"]
```

I have read that "there is no shortcircuiting of booleans, all conditions will be evaluated" [here](https://cuetorials.com/overview/expressions/#conditional-fields) so I'm not sure if there's a solution to my issue. Creating an opposing `if` expression like `if c == "tacoma" {""}` is not desirable either since I want one less entry and not an empty string, null, etc.

**Comment by slewiskelly** [comment, marked as answer]
You need:

```cue
cars: ["tacoma", "escort"]

cars2: [ for c in cars if c != "tacoma" {c}]
```

Here is an example of excluding multiple:

```cue
import "list"

cars: ["tacoma", "escort", "delorean"]

cars2: [ for c in cars if !list.Contains(_exclude, c) {c}]

_exclude: ["tacoma", "escort"]
```
